modules/vision/src/pose-estimation/vpPoseDementhon.cpp

```diff
@@ -466,9 +469,34 @@ void vpPose::poseDementhonPlan(vpHomogeneousMatrix &cMo)
   }
   vpColVector sv;
   vpMatrix Ap, imA, imAt, kAt;
-  int irank = A.pseudoInverse(Ap, sv, 1.e-6, imA, imAt, kAt);
-  if (irank != 3) {
-    throw(vpException(vpException::fatalError, "In Dementhon planar, rank is not 3"));
+  bool isRankEqualTo3(false); // Indicates if the rank of A is the expected one
+  double logNofSvdThresh = std::log(dementhonSvThresh)/lnOfSvdFactorUsed; // Get the log_n(dementhonSvThresh), where n is the factor by which we will multiply it if the svd decomposition fails.
```

Comment:
Maybe CI will break when using c++98: https://cdash-ci.inria.fr/index.php?project=ViSP
If this is the case, we will have to replace with `bool isRankEqualTo3 = false;` and below `int irank = 0;`

Reply:
Sorry, did not think of being c++98 compatible.
Changed in commit 279c1e7acb7e675682c481644b95263774853f0a , lines 472 and 476